ubuntu-deploy/index.md

```diff
@@ -0,0 +1,414 @@
+    proxy_set_header Connection 'upgrade';
+    proxy_set_header Host $host;
+    proxy_cache_bypass $http_upgrade;
+  }
+}
+```
+
+This simply forward the request to the domain to the Directus application running at `localhost` port `8055`.
+
+Let's step through the config file:
+
+- listen 80: Listens for incoming connections on port 80, the default for HTTP traffic.
+
+- server_name directus.exampledomain.com: Matches requests to this domain.
+
+- location / { ... }: Handles all requests for directus.exampledomain.com.
+
+  - proxy_pass `http://localhost:8055`: Forwards requests to the Directus application.
+
+  - proxy_http_version 1.1: Uses the HTTP/1.1 protocol for Nginx-proxy communication.
+
+  - proxy_set_header Upgrade $http_upgrade: Essential for WebSocket connections.
+
+  - proxy_set_header Connection 'upgrade': Indicates connection upgrade.
+
+  - proxy_set_header Host $host: Sends original host information to the server.
+
+  - proxy_cache_bypass $http_upgrade: Bypasses caching for WebSockets.
+
+To test the Nginx configuration files for syntax errors, you can use the following command:
+
+```bash
+sudo nginx -t
+```
+
+Create a symbolic link to enable the site:
+
+```bash
+sudo ln -s /etc/nginx/sites-available/directus /etc/nginx/sites-enabled
+```
+
+And restart Nginx:
+
+```bash
+sudo systemctl restart nginx
+```
+
+Now you should be able to access your Directus application without adding the port at `http://directus.exampledomain.com`.
+
+![Directus application accessed at the domain but insecure][image-1]
+
+## Securing Your Application with SSL (Optional but Recommended)
+
+Implementing SSL (Secure Sockets Layer) encryption is crucial for safeguarding data transmitted between your users and the server. [Let's Encrypt][let-encrypt] offers free SSL certificates, and here's how to set it up for your Directus application:
+
+1. Install Certbot: On your server, run the following commands to install Certbot and the Certbot Nginx plugin:
+
+```bash
+sudo apt install certbot python3-certbot-nginx -y
+```
+
+2. Obtain an SSL Certificate: Run Certbot with the --nginx option, specifying your domain (directus.exampledomain.com in this case):
+
+```bash
+sudo certbot --nginx -d directus.exampledomain.com
+```
+
+Certbot will interactively guide you through the setup process.
+
+Ensure you select the option to redirect HTTP traffic to HTTPS when prompted. Certbot will automatically configure Nginx to use the SSL certificate.
+
+![HTTP to HTTPS redirect with certbot][image-9]
+
+:::info Renew certificate
+
+Also, ensure to renew the certificate before it expires to maintain a secure connection.
+
+:::
+
+3. Verify SSL Configuration: After the setup is complete, visit your Directus application using `https://directus.exampledomain.com` in a web browser. You should see a padlock icon indicating a secure SSL connection. You should also be automatically redirected from `http` to `https`.
+
+![Secured Directus application][image-3]
```

Comment:
I'll crop this when I upload it to remove whitespace below the popup, but really useful.